openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/OpenshiftHelmLintMojo.java

```diff
@@ -51,9 +44,4 @@ protected File getKubernetesTemplate() {
   protected HelmConfig.HelmType getDefaultHelmType() {
     return HelmConfig.HelmType.OPENSHIFT;
   }
-
-  @Override
-  protected String getLogPrefix() {
-    return OpenShift.DEFAULT_LOG_PREFIX;
-  }
```

Comment:
The OpenShift overrides are required to preserve the OpenShift-specific behavior (in this case the logged statements should be prefixed with `oc: ` as opposed to `k8s: '  same happens with the other apparently redundant overrides.

Reply:
Ah, I missed that this reads `OpenShift` :) I reverted this change in c03626830525443607c4cccd0715228d76d57c2a.

Reply:
As for the other apparently redundant overrides: I feel that they are actually redundant. (https://github.com/eclipse/jkube/pull/2714#discussion_r1498762758 for example.) But of course I might be wrong 😀

Reply:
https://github.com/eclipse/jkube/pull/2714#discussion_r1511430449

Basically there are 4 things to override for the OpenShift Helm experience.
- Static literal for the logger
- Default enum value for the helm type
- `getKubernetesManifest`: should respond to the user-provided property or configuration  `openShiftManifest` instead
- getKubernetesTemplate: should respond to the user-provided property or configuration  `openshiftTemplate` instead (note that [there is a bug in the current OpenshiftHelmPushMojo](https://github.com/eclipse/jkube/blob/5da6068d7756784aef9569568ee80f98da97e296/openshift-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/OpenshiftHelmPushMojo.java#L36) in the property, it reads `"jkube.kubernetesManifest"` when it should be `"jkube.openshiftTemplate""

As for the last two, I assume they were introduced to be able to override the default value of the parameter.

The only one redundant or unnecessary for the Helm Push mojo is the one for the kubernetesManifest, which is used for the warning log. (As a user, I want to get warned if I run the helm goal without having run the resource goal before, so that I'm aware of my misconfiguration). However, this warning should also apply for the Lint mojo.

The new `AbstractHelmMojo` has `kubernetesTemplate`, so this override should be kept.

Reply:
This holds for all OpenShift Helm classes that are changed, so for:

- OpenshiftHelmLintMojo
- OpenshiftHelmMojo
- OpenshiftHelmPushMojo

Let's be methodical and list per method-to-be-overridden here:

### getKubernetesManifest()

- OpenshiftHelmLintMojo: not available in parent classes; can't override
- OpenshiftHelmMojo: overridden on line 44: https://github.com/eclipse/jkube/pull/2714/files#diff-6290a43bd73acd81a35c9ece497c1315ed7e146c5e52a39887ccda6f40c29868R44
- OpenshiftHelmPushMojo: not available in parent classes; can't override

The `getKubernetesManifest()` method is defined in `org.eclipse.jkube.maven.plugin.mojo.build.HelmMojo` on line 70: https://github.com/eclipse/jkube/pull/2714/files#diff-1e94b9a102c6e2c667d4ee4da8631008a2237cb933ceff7bb0d365fe945b6fc0R70. Because it's available in `HelmMojo`, it can be overridden in `OpenshiftHelmMojo`.
Put differently; because it is not available in `org.eclipse.jkube.maven.plugin.mojo.build.AbstractHelmMojo`, it can not be overridden in `HelmLintMojo` or in `OpenshiftHelmLintMojo`, and it can not be overridden in `HelmPushMojo` or in `OpenshiftHelmPushMojo`.

The Kubernetes manifest property is only used in `HelmMojo` to generate a warning when `k8s:resource` or `oc:resource` is not ran. Do you think that the same warning should be shown when `k8s:helm-lint`, `oc:helm-lint`, `k8s:helm-push` or `oc:helm-push` is ran?

### getKubernetesTemplate()

- OpenshiftHelmLintMojo: overridden on line 40: https://github.com/eclipse/jkube/pull/2714/files#diff-814cb60fbdb969f6117757d510d1ee3502387b6cce13fd4bd654f9014ee195a6R40
- OpenshiftHelmMojo: overridden on line 50: https://github.com/eclipse/jkube/pull/2714/files#diff-6290a43bd73acd81a35c9ece497c1315ed7e146c5e52a39887ccda6f40c29868R50
- OpenshiftHelmPushMojo: overridden on line 39: https://github.com/eclipse/jkube/pull/2714/files#diff-e8f5cf8fb0ab049afb61be5e7a00aacefd5bbc68009345c6ba1e86806eecc4b2R39

### getDefaultHelmType()

- OpenshiftHelmLintMojo: overridden on line 45: https://github.com/eclipse/jkube/pull/2714/files#diff-814cb60fbdb969f6117757d510d1ee3502387b6cce13fd4bd654f9014ee195a6R45
- OpenshiftHelmMojo: overridden on line 54: https://github.com/eclipse/jkube/pull/2714/files#diff-6290a43bd73acd81a35c9ece497c1315ed7e146c5e52a39887ccda6f40c29868R54
- OpenshiftHelmPushMojo: overridden on line 44: https://github.com/eclipse/jkube/pull/2714/files#diff-e8f5cf8fb0ab049afb61be5e7a00aacefd5bbc68009345c6ba1e86806eecc4b2R44

### getLogPrefix()

- OpenshiftHelmLintMojo: overridden on line 50: https://github.com/eclipse/jkube/pull/2714/files#diff-814cb60fbdb969f6117757d510d1ee3502387b6cce13fd4bd654f9014ee195a6R50
- OpenshiftHelmMojo: overridden on line 59: https://github.com/eclipse/jkube/pull/2714/files#diff-6290a43bd73acd81a35c9ece497c1315ed7e146c5e52a39887ccda6f40c29868R59
- OpenshiftHelmPushMojo: overridden on line 49: https://github.com/eclipse/jkube/pull/2714/files#diff-e8f5cf8fb0ab049afb61be5e7a00aacefd5bbc68009345c6ba1e86806eecc4b2R49

Reply:
This sounds about right.

> The Kubernetes manifest property is only used in `HelmMojo` to generate a warning when `k8s:resource` or `oc:resource` is not ran. Do you think that the same warning should be shown when `k8s:helm-lint`, `oc:helm-lint`, `k8s:helm-push` or `oc:helm-push` is ran?

This is necessary for those goals that depend on the resources. So only for the initial HelmMojo. However, for the others, it'd be nice to have a similar warning reminding user to run the helm goal first (follow-up PR), the Gradle task behavior should be aligned too (follow-up PR 2).